keras_retinanet/preprocessing/open_images.py

```diff
@@ -0,0 +1,206 @@
+                                            'Confidence', 'XMin', 'XMax', 'YMin',
+                                            'YMax'])
+        reader.next()
+        for _ in reader:
+            cnt += 1
+
+        print ('There are {} lines in subset: {}'.format(cnt, subset))
+
+    id_annotations = dict()
+
+    with open(annotations_path, 'r') as csvfile:
+        reader = csv.DictReader(csvfile,
+                                fieldnames=['ImageID', 'Source', 'LabelName',
+                                            'Confidence', 'XMin', 'XMax', 'YMin',
+                                            'YMax'])
+        reader.next()
+
+        images_sizes = dict()
+        for line, row in enumerate(reader):
+            frame = row['ImageID']
+
+            class_name = row['LabelName']
+
+            if class_name not in cls_index_dict:
+                continue
+
+            cls_id = cls_index_dict[class_name]
+
+            img_path = os.path.join(main_dir, 'images', subset, frame + '.jpg')
+            if frame in images_sizes:
+                width, height = images_sizes[frame]
+            else:
+                try:
+                    with Image.open(img_path) as img:
+                        width, height = img.width, img.height
+                        images_sizes[frame] = (width, height)
+                except Exception:
+                    continue
+
+            x1 = float(row['XMin'])
+            x2 = float(row['XMax'])
+            y1 = float(row['YMin'])
+            y2 = float(row['YMax'])
+
+            x1_abs = int(round(x1 * width))
+            x2_abs = int(round(x2 * width))
+            y1_abs = int(round(y1 * height))
+            y2_abs = int(round(y2 * height))
+
+            # Check that the bounding box is valid.
+            if x2 <= x1:
+                raise ValueError('line {}: x2 ({}) must be higher than x1 ({})'.format(line, x2, x1))
+            if y2 <= y1:
+                raise ValueError('line {}: y2 ({}) must be higher than y1 ({})'.format(line, y2, y1))
+
+            if y2_abs == y1_abs:
+                print ('filtering line {}: rounding y2 ({}) and y1 ({}) makes them equal'.format(line, y2, y1))
+                continue
+
+            if x2_abs == x1_abs:
+                print ('filtering line {}: rounding x2 ({}) and x1 ({}) makes them equal'.format(line, x2, x1))
+                continue
+
+            img_id = row['ImageID']
+
+            annotation = {'cls_id': cls_id, 'x1': x1, 'x2': x2, 'y1': y1, 'y2': y2}
+
+            if img_id in id_annotations:
+
+                annotations = id_annotations[img_id]
+                annotations['boxes'].append(annotation)
+            else:
+                id_annotations[img_id] = {'w': width, 'h': height, 'boxes': [annotation]}
+    return id_annotations
+
+
+class OpenImagesGenerator(Generator):
+    def __init__(
+            self, main_dir, subset, version, labels_filter,
```

Comment:
`version` could have the same default here as in `train.py`. As far as I understood it, that default makes sense?

Reply:
I added a default value for version, corresponding to the V3 of the dataset. And for labels_filter too, since by default we want to train on the whole dataset.